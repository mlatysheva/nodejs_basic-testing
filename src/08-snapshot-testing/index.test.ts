import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    const links = ['users', 'posts', 'create post'];
    const generatedNavList = generateLinkedList(links);
    const mockedNavList = {
      value: 'users',
      next: {
        value: 'posts',
        next: {
          value: 'create post',
          next: {
            value: null,
            next: null
          }
        }
      }
    }
    expect (generatedNavList).toStrictEqual(mockedNavList);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const links = ['users', 'posts', 'create post'];
    const generatedNavList = generateLinkedList(links);
    expect(generatedNavList).toMatchSnapshot();
  });
});
