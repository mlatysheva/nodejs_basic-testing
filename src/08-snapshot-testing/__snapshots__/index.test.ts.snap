// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "create post",
    },
    "value": "posts",
  },
  "value": "users",
}
`;
